### ISSUE TITLE
fcct crashes on s390x with "panic: runtime error: invalid memory address or nil pointer dereference"

### ISSUE THREAD

**Issue by Prashanth684** [coreos/fcct#142]
On s390x noticed that fcct crashes with this backtrace:

goroutine 1 [running]:
panic(0x2aa050a0a00, 0x2aa0539aa60)
	/usr/lib/golang/src/runtime/panic.go:1064 +0x51c fp=0xc000198e98 sp=0xc000198de0 pc=0x2aa04c22bac
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.handleErr(0xc000199d38)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/yaml.go:276 +0xae fp=0xc000198eb8 sp=0xc000198e98 pc=0x2aa04dc501e
runtime.call32(0x0, 0x2aa05111ec8, 0xc000146048)
	/usr/lib/golang/src/runtime/asm_s390x.s:464 +0x94 fp=0xc000198ee0 sp=0xc000198eb8 pc=0x2aa04c533b4
runtime.reflectcallSave(0xc000199030, 0x2aa05111ec8, 0xc000146048, 0x8051126d0)
	/usr/lib/golang/src/runtime/panic.go:881 +0x50 fp=0xc000198f08 sp=0xc000198ee0 pc=0x2aa04c225c0
runtime.runOpenDeferFrame(0xc000000180, 0xc000146000, 0xc000199068)
	/usr/lib/golang/src/runtime/panic.go:855 +0x34e fp=0xc000198fb0 sp=0xc000198f08 pc=0x2aa04c2245e
panic(0x2aa050a0a00, 0x2aa0539aa60)
	/usr/lib/golang/src/runtime/panic.go:969 +0x132 fp=0xc000199068 sp=0xc000198fb0 pc=0x2aa04c227c2
runtime.panicmem(...)
	/usr/lib/golang/src/runtime/panic.go:212
runtime.sigpanic()
	/usr/lib/golang/src/runtime/signal_unix.go:695 +0x3ea fp=0xc000199090 sp=0xc000199068 pc=0x2aa04c3b50a
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.yaml_parser_parse_node(0xc00013a000, 0xc00013a290, 0x100000000000080, 0x8)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/parserc.go:563 +0x496 fp=0xc000199aa8 sp=0xc000199098 pc=0x2aa04dad6b6
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.yaml_parser_state_machine(0xc00013a000, 0xc00013a290, 0x3ffaba401a0)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/parserc.go:170 +0x1a4 fp=0xc000199ad0 sp=0xc000199aa8 pc=0x2aa04dabf34
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.yaml_parser_parse(0xc00013a000, 0xc00013a290, 0x3ffabb3ae98)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/parserc.go:129 +0x82 fp=0xc000199af0 sp=0xc000199ad0 pc=0x2aa04dabd42
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.(*parser).peek(0xc00013a000, 0x0)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/decode.go:102 +0x3c fp=0xc000199b10 sp=0xc000199af0 pc=0x2aa04da39fc
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.(*parser).parse(0xc00013a000, 0x0)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/decode.go:141 +0x36 fp=0xc000199b60 sp=0xc000199b10 pc=0x2aa04da3ca6
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.(*parser).parseChild(0xc00013a000, 0xc000132460, 0x0)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/decode.go:186 +0x2a fp=0xc000199bb8 sp=0xc000199b60 pc=0x2aa04da437a
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.(*parser).document(0xc00013a000, 0x300000000000030)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/decode.go:195 +0xa6 fp=0xc000199c10 sp=0xc000199bb8 pc=0x2aa04da4526
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.(*parser).parse(0xc00013a000, 0x0)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/decode.go:151 +0x13a fp=0xc000199c60 sp=0xc000199c10 pc=0x2aa04da3daa
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.unmarshal(0xc000136000, 0xf4, 0x600, 0x2aa05066540, 0xc000118120, 0x2aa0511d6c0, 0x0, 0x0)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/yaml.go:161 +0x22c fp=0xc000199d00 sp=0xc000199c60 pc=0x2aa04dc4b1c
github.com/coreos/fcct/vendor/gopkg.in/yaml%2ev3.Unmarshal(...)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/vendor/gopkg.in/yaml.v3/yaml.go:89
github.com/coreos/fcct/config.Translate(0xc000136000, 0xf4, 0x600, 0xc000136000, 0x0, 0x0, 0x0, 0x0, 0x2f, 0x2aa04c6eed0, ...)
	/builddir/build/BUILD/fcct-0.6.0/_build/src/github.com/coreos/fcct/config/config.go:60 +0x6a fp=0xc000199dc8 sp=0xc000199d00 pc=0x2aa04f96c6a
main.main()
	/builddir/build/BUILD/fcct-0.6.0/internal/main.go:93 +0x53e fp=0xc000199f80 sp=0xc000199dc8 pc=0x2aa04f977fe
runtime.main()
	/usr/lib/golang/src/runtime/proc.go:203 +0x236 fp=0xc000199fd8 sp=0xc000199f80 pc=0x2aa04c25956
runtime.goexit()
	/usr/lib/golang/src/runtime/asm_s390x.s:779 +0x2 fp=0xc000199fd8 sp=0xc000199fd8 pc=0x2aa04c55282

This looks very similar to https://github.com/golang/go/issues/38195 which should have been fixed in newer versions of go 1.14 and go 1.13.

**Comment by bgilbert** [comment]
Where did you get your FCCT binary?

**Comment by Prashanth684** [comment]
the one that's part of f32
```
[coreos-assembler]$ rpm -q --whatprovides /usr/bin/fcct
fedora-coreos-config-transpiler-0.6.0-1.fc32.s390x
```

**Comment by Prashanth684** [comment]
I tried building fcct with go 1.3.8 which did not have the fix. It crashed with the backtrace above. I tried it with 1.13.15 a later release which had the fix and it worked without issues. Not sure what version of go is used to generate the one that's part of f32

**Comment by bgilbert** [comment]
Just to confirm, could you try building with the Go compiler packaged in F32?  We'll be releasing a new version of FCCT shortly, and that should pick up any fixes in the current F32 compiler.

**Comment by Prashanth684** [comment]
ah..yes that works. the version of go is 1.14.9 on f32. isn't fcct compiled with the version of go matching f32 ?

**Comment by bgilbert** [comment]
Yes, but the 0.6.0-1.fc32 package was built on May 29 with the Go version that was in F32 at the time.

**Comment by bgilbert** [comment]
This should be fixed by [0.7.0-1.fc32](https://bodhi.fedoraproject.org/updates/FEDORA-2020-75a4e2a40c).  Thanks for reporting.
